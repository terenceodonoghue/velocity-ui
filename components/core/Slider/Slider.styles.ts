import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { rgba } from 'polished';

export const sliderInput: FunctionInterpolation<Theme> = ({
  palette,
  transitions,
}) => css`
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 100%;
  background-color: ${rgba(palette.accent, 0.15)};
  outline: none;
  transition: opacity ${transitions.duration.standard}ms
    ${transitions.easing.sharp};

  &::-moz-range-progress {
    background-color: ${palette.accent};
  }

  &::-moz-range-thumb {
    border: solid 4px ${palette.accent};
    border-radius: 50%;
    height: 8px;
    width: 8px;
    cursor: pointer;
  }

  &::-webkit-slider-thumb {
    appearance: none;
    border: solid 4px ${palette.accent};
    border-radius: 50%;
    height: 16px;
    width: 16px;
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.3;
  }
`;

export const sliderText: FunctionInterpolation<Theme> = ({ palette }) => css`
  display: flex;

  p {
    line-height: 1.467em;
    margin: 0;
  }

  p:nth-of-type(1) {
    flex: 1;
  }

  p:nth-of-type(2) {
    color: ${palette.neutral[600]};
  }
`;
