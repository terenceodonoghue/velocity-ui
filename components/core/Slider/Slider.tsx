import React, { forwardRef } from 'react';
import * as css from './Slider.styles';
import type { SliderProps } from './Slider.types';

const Slider = forwardRef<HTMLInputElement, SliderProps>(
  ({ primary, secondary, ...props }, ref) => (
    <label data-testid="slider">
      {(primary || secondary) && (
        <div css={css.sliderText}>
          {primary && <p data-testid="slider-primary">{primary}</p>}
          {secondary && <p data-testid="slider-secondary">{secondary}</p>}
        </div>
      )}
      <input
        css={css.sliderInput}
        data-testid="slider-input"
        ref={ref}
        type="range"
        {...props}
      />
    </label>
  ),
);

export default Slider;
