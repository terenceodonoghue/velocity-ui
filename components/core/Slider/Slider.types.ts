import { InputHTMLAttributes } from 'react';

export interface SliderProps extends InputHTMLAttributes<HTMLInputElement> {
  primary?: string;
  secondary?: string;
}
