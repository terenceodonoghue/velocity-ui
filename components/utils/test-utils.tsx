import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
