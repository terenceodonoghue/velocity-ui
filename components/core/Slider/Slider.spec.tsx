import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import Slider from './Slider';

describe('Slider', () => {
  it('renders primary label', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId, queryByTestId } = render(
      <Slider primary={content} />,
    );

    expect(getByTestId('slider-primary')).toHaveTextContent(content);
    expect(queryByTestId('slider-secondary')).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders secondary label', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId, queryByTestId } = render(
      <Slider secondary={content} />,
    );

    expect(queryByTestId('slider-primary')).toBeNull();
    expect(getByTestId('slider-secondary')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });

  it('renders primary and secondary labels', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(
      <Slider primary={content} secondary={content} />,
    );

    expect(getByTestId('slider-primary')).toHaveTextContent(content);
    expect(getByTestId('slider-secondary')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
