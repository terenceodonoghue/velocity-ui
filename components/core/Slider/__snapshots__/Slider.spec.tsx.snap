// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders primary and secondary labels 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 p {
  line-height: 1.467em;
  margin: 0;
}

.emotion-0 p:nth-of-type(1) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 p:nth-of-type(2) {
  color: #8798ad;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 100%;
  background-color: rgba(46,91,255,0.15);
  outline: none;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-1::-moz-range-progress {
  background-color: #2e5bff;
}

.emotion-1::-moz-range-thumb {
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 8px;
  width: 8px;
  cursor: pointer;
}

.emotion-1::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  cursor: pointer;
}

.emotion-1:disabled {
  opacity: 0.3;
}

<div>
  <label
    data-testid="slider"
  >
    <div
      class="emotion-0"
    >
      <p
        data-testid="slider-primary"
      >
        You can't input the array without navigating the redundant GB alarm!
      </p>
      <p
        data-testid="slider-secondary"
      >
        You can't input the array without navigating the redundant GB alarm!
      </p>
    </div>
    <input
      class="emotion-1"
      data-testid="slider-input"
      type="range"
    />
  </label>
</div>
`;

exports[`Slider renders primary label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 p {
  line-height: 1.467em;
  margin: 0;
}

.emotion-0 p:nth-of-type(1) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 p:nth-of-type(2) {
  color: #8798ad;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 100%;
  background-color: rgba(46,91,255,0.15);
  outline: none;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-1::-moz-range-progress {
  background-color: #2e5bff;
}

.emotion-1::-moz-range-thumb {
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 8px;
  width: 8px;
  cursor: pointer;
}

.emotion-1::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  cursor: pointer;
}

.emotion-1:disabled {
  opacity: 0.3;
}

<div>
  <label
    data-testid="slider"
  >
    <div
      class="emotion-0"
    >
      <p
        data-testid="slider-primary"
      >
        The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
      </p>
    </div>
    <input
      class="emotion-1"
      data-testid="slider-input"
      type="range"
    />
  </label>
</div>
`;

exports[`Slider renders secondary label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 p {
  line-height: 1.467em;
  margin: 0;
}

.emotion-0 p:nth-of-type(1) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 p:nth-of-type(2) {
  color: #8798ad;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 100%;
  background-color: rgba(46,91,255,0.15);
  outline: none;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-1::-moz-range-progress {
  background-color: #2e5bff;
}

.emotion-1::-moz-range-thumb {
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 8px;
  width: 8px;
  cursor: pointer;
}

.emotion-1::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  cursor: pointer;
}

.emotion-1:disabled {
  opacity: 0.3;
}

<div>
  <label
    data-testid="slider"
  >
    <div
      class="emotion-0"
    >
      <p
        data-testid="slider-secondary"
      >
        You can't quantify the interface without parsing the neural ADP feed!
      </p>
    </div>
    <input
      class="emotion-1"
      data-testid="slider-input"
      type="range"
    />
  </label>
</div>
`;
